src/Azure/Orleans.Streaming.AzureStorage/Hosting/ClientBuilderExtensions.cs

```diff
@@ -11,25 +10,24 @@ public static class ClientBuilderExtensions
         /// <summary>
         /// Configure cluster client to use azure queue persistent streams. 
         /// </summary>
-        public static IClientBuilder AddAzureQueueStreams<TDataAdapter>(this IClientBuilder builder,
+        public static IClientBuilder AddAzureQueueStreams(this IClientBuilder builder,
             string name,
-            Action<ClusterClientAzureQueueStreamConfigurator<TDataAdapter>> configure)
-            where TDataAdapter : IAzureQueueDataAdapter
+            Action<IClusterClientAzureQueueStreamConfigurator> configure)
         {
             //the constructor wires up DI with AzureQueueStream, so has to be called regardless configure is null or not
-            var configurator = new ClusterClientAzureQueueStreamConfigurator<TDataAdapter>(name, builder);
+            var configurator = new ClusterClientAzureQueueStreamConfigurator(name, builder);
             configure?.Invoke(configurator);
+            configurator.ConfigureDefaults();
```

Comment:
should the defaults be before configure.invoke?

Reply:
Not the way I'm configuring them.  Basically the defaults configure required components with their defaults -if- they've not already been configured.

I'm not terribly happy with this solution, but not come up with a better approach, as this requires creators of a configurator to understand the expected order, and if they forget or get it wrong the configurator may not function correctly.  I'd prefer a pattern that's less prone to user error.

Open to suggestions.

Reply:
Dropped ConfigureDefaults in favor of a default service (rather than a named service).  If a named service is not found for a provider, it will fail over to the global default.  Application developers can override this default by registering it in the container before or after the configuring azure streams.
Will show up in next push.